py/server/deephaven/parquet.py

```diff
@@ -380,34 +409,20 @@ def write_partitioned(
             for_read=False,
             generate_metadata_files=generate_metadata_files,
             base_name=base_name,
+            table_definition=table_definition,
+            col_definitions=col_definitions,
+            index_columns=index_columns,
         )
-
-        table_definition = None
-        if col_definitions is not None:
-            table_definition = _JTableDefinition.of([col.j_column_definition for col in col_definitions])
-
-        if index_columns:
-            index_columns_array = _j_array_of_array_of_string(index_columns)
-            if table_definition:
-                _JParquetTools.writeKeyValuePartitionedTable(table.j_object, table_definition, destination_dir,
-                                                             write_instructions, index_columns_array)
-            else:
-                _JParquetTools.writeKeyValuePartitionedTable(table.j_object, destination_dir, write_instructions,
-                                                             index_columns_array)
-        else:
-            if table_definition:
-                _JParquetTools.writeKeyValuePartitionedTable(table.j_object, table_definition, destination_dir,
-                                                             write_instructions)
-            else:
-                _JParquetTools.writeKeyValuePartitionedTable(table.j_object, destination_dir, write_instructions)
+        _JParquetTools.writeKeyValuePartitionedTable(table.j_object, destination_dir, write_instructions)
     except Exception as e:
         raise DHError(e, "failed to write to parquet data.") from e
 
 
 def batch_write(
     tables: List[Table],
     paths: List[str],
-    col_definitions: List[Column],
+    table_definition: Optional[Union[Dict[str, DType], List[Column]]] = None,
+    col_definitions: Optional[List[Column]] = None,
```

Comment:
change `table_definition` to 'col_definition` and remove the old col_definition?

Reply:
My motivation behind making the change from `col_definitions` to `table_definition` was:
1.  The Java APIs all accept a tableDefinition, so I wanted to keep it consistent here.
2. The `read` call in the python code accepts `table_definition` but `write` call accepts `col_definitions`. So wanted to make it consistent.

So I have tried to deprecate the `col_definitions` argument and add a `table_definition`.